### PR TITLE
bpo-33944: Add site.py debugging via PYTHONSITEDEBUG

### PR DESCRIPTION
**Pull request by native-api**
https://bugs.python.org/issue33944

### CODE REVIEW

File: Lib/site.py [hunk 1]
```diff
@@ -74,6 +74,12 @@
 import builtins
 import _sitebuiltins
 
+
+if os.environ.get("PYTHONSITEDEBUG", ""):
+    import pdb
+    pdb.set_trace()
```
Comment (methane) [inline Lib/site.py]:
```suggestion
    breakpoint()
```

File: Lib/site.py [hunk 2]
```diff
@@ -74,6 +74,11 @@
 import builtins
 import _sitebuiltins
 
+
+if os.environ.get("PYTHONSITEDEBUG", ""):
```
Comment (tiran) [inline Lib/site.py:78]: Any PYTHON env var must be ignored when ignore environment is set.

```suggestion
if not sys.flags.ignore_environment and os.environ.get("PYTHONSITEDEBUG"):
```

File: Doc/using/cmdline.rst [hunk 3]
```diff
@@ -917,6 +917,15 @@ conflict.
       See :pep:`540` for more details.
 
 
+.. envvar:: PYTHONSITEDEBUG
+
+   If set to a non-empty string, drop into the :mod:`pdb` prompt
+   at the start of the :mod:`site` module.
+   This is useful to diagnose issues in 3rd-party code that runs at startup.
+
+   .. versionadded:: 3.8
```
Comment (tiran) [inline Doc/using/cmdline.rst:926]:
```suggestion
   .. versionadded:: 3.10
```

**Review by tiran — CHANGES_REQUESTED** [review]
This PR piggybacks on BPO 33944. IMO it needs a separate BPO and a more thorough discussion.

**Comment by erlend-aasland** [comment]
Closed as per https://github.com/python/cpython/pull/12107#pullrequestreview-431773752

**Pull request closed without merging**